services/ui/src/lib/components/VariableDiag/VariableDiag.svelte

```diff
@@ -1,16 +1,28 @@
 <script>
+  import { extent, max, min } from "d3";
+
   import Observations from "./Observations.svelte";
 
   export let title;
   export let guess;
   export let analysis;
+
+  let domain = [0, 1];
+  let range = [0, 1];
+
+  $: if (guess && analysis) {
+    const bins = guess.bins.concat(analysis.bins);
+
+    domain = [min(bins, (d) => d.lower), max(bins, (d) => d.upper)];
+    range = extent(bins, (d) => d.value);
+  }
 </script>
 
 <article class="flow">
   <h2>{title}</h2>
   <div data-layout="cluster">
-    <Observations title="Observation - Background" {...guess} />
-    <Observations title="Observation - Analysis" {...analysis} />
+    <Observations title="Observation - Background" {...guess} {domain} {range} />
```

Comment:
Triple checking my JS knowledge. Am I correct in thinking:

* That `{...guess}` is a spread operator and expands into `observations`, `mean`, `std`, and `bins` that `Observations.svelte` consumes?
* And that `guess` is populated by the JSON object from the backend? (by way of `...diag` passed into the `VariableDiag` class in `index.svelte`?)

Reply:
That’s correct. It’s the same as writing

```
<Observations mean={guess.mean} std={guess.std} and bins={guess.bins} />
```